src/foam/comics/v2/Clickable.js

```diff
@@ -0,0 +1,51 @@
+/**
+* @license
+* Copyright 2023 The FOAM Authors. All Rights Reserved.
+* http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+foam.CLASS({
+  package: 'foam.comics.v2',
+  name: 'Clickable',
+
+  imports: [
+    'click?',
+    'config?',
+    'dblclick?',
+    'selection?'
+  ],
+
+  methods: [
+    function insertClick(idObj /* expects an object with an id prop*/) {
+      var self = this;
+      this.callIf( this.dblclick && ! this.config.disableSelection, function() {
+        this.on('dblclick', function(evt) {
+            if ( self.shouldEscapeEvts(evt) ) return;
+            self.dblclick.call(self, null, idObj.id);
+        });
+      })
+      .callIf( this.click && ! this.config.disableSelection, function() {
+        this.on('click', function(evt) {
+          if ( self.shouldEscapeEvts(evt) ) return;
+          self.selection = idObj.id;
+          self.click.call(self, null, idObj.id);
+        });
+      });
+    }
+  ],
+  listeners: [
+    {
+      name: 'shouldEscapeEvts',
```

Comment:
Should this be a listener?

Reply:
Rather than just a regular method, since it is synchronous and has a return value and is never called as a callback.